pkg/impersonation/impersonation.go

```diff
@@ -47,6 +52,8 @@ func New(userInfo user.Info, clusterContext *config.UserContext) (Impersonator,
 	return impersonator, nil
 }
 
+// SetUpImpersonation creates a service account on a cluster with a clusterrole and binding allowing it to impersonate a Rancher user.
+// Returns a reference to the service account which can be used to retrieve the account token, or an error if creating any of the resources failed.
```

Comment:
Not sure that `which can be used to retrieve the account token` is necessary here. Alternatively, if you want to include it, maybe also include a reference to the below `GetToken` function which you would use to get the token. 

Something like:
```
Returns a reference to the service account (can be used by GetToken to retrieve its token), or an error if creating any of the resources failed.
```

Reply:
done